### ISSUE TITLE
Heat damage is exceedingly powerful when compared to other special damage types

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#5976]
**Is your feature request related to a problem? Please describe.**
Heat damage is an extremely powerful form of damage that doesn't lose much effectiveness, if any, as you increase in tier. A small secondary weapon unlocked during the FW campaign, the flamethrower, is one of the best weapons for punching up and taking down larger ships. (Not on its own, of course, and to an extend that's the whole purpose of the weapon given how the FW use it to take down Navy capitals, but the degree to which it is successful is a bit much.)
Other special damage types by comparison aren't nearly as effective. Disruption is probably the second best in its usefulness in disabling ships without destroying them, slowing being more of an annoyance, and ionization being a worse version of heat damage in terms of its utility as a special damage type.

Something should probably be done about heat damage, either in looking to alter the effectiveness of the weapons that use it (mainly the flamethrower), or in changing how heat damage itself works.

**Related Issue Links**
Idk.

**Describe the solution(s) you'd like**
The following are a number of things I've seen discussed in relation to this issue:
* Have overheating a ship not outright disable it, but instead give it a chance to become disabled for a brief moment, akin to low crew behavior. This would attack heat damage's major strength of completely taking a ship out of combat. Ion damage can also do this, but it is much more difficult to achieve, and often times you only prevent a ship from using some of its systems instead of outright disabling it, so at least a heavily ionized ship can fight back to some degree or try to run if it doesn't have energy for its weapons.
* Just nerf the flamethrower's heat damage. I don't know by how much, but nerf it.
* Buff ion damage. As described, ion damage is the closest damage type to heat damage, but worse. This would probably occur in conjunction with a nerf to heat damage so that we aren't just creating two overpowered damage types. Buffing ion damage might entail simply increasing ion damage on all ion weapons. Another idea that came up though that I like was allowing ion damage to bypass the 50% debuff to special damage that shields give, so while ion damage is always effective against ships, heat damage is more effective against de-shielded ships. This could also create some inter-play between these two damage types; hit a ship with ion damage to help sap its shields then hit it with heat damage to practically disable it when the shields are down.

**Additional context**
Not really a way to nerf heat damage (as the main concern with heat damage is its effectiveness against larger ships, but this idea would actually buff it against larger ships and nerf it against smaller ships), but perhaps heat damage should scale with the ship it is hitting just like hit force does. Hit force use to affect ships linearly with their mass; double the mass, receive half the impact from hit force. Hit force has since been changed to bring larger and smaller ships closer together in terms of how they are affected by hit force. Perhaps the same should be done with heat damage, as currently larger ships are affected less by heat damage than smaller ships given that larger ships have more mass to heat up.

A related topic that came up when discussing this was changing how shield leakage works. Currently if your shields are up, all incoming special damage types deal half damage. This leads to quite a drastic jump in damage after your shields are down. I believe that shield leakage should either scale with the amount of shields remaining (e.g. 50% resistance at 100% shields and 25% resistance at 50% shields, `double leakage = 1. - .5 * shields / attributes.Get("shields");`) and/or the resistance that shields provide should be lowered (by at least half). Or a more extreme solution, active shields cutting special damage down should be removed entirely, as we now have special resistance types for each damage type anyway, although I'm less keen on this idea and would actually rather like to see how shields affect incoming damage be customizable by outfits (e.g. having an shield generator that specifically cuts down incoming heat damage while active but not other damage types), but that's getting into a separate issue entirely.

**Comment by thurokiir** [comment]
Star Sector has a similar balance property "flux" which is the accumulation of heat over the course of battle.

The interaction with flux is that in order to drop the heat the ship must stop all function, and dump the flux.

The balance of flux is that the variables the players are provided: Rate of Flux Dissipation, Capacity of Flux, Passive Flux Dispersion and a differentiation between 'Hard' and 'Soft' flux.

Hard Flux can ONLY be dissipated by a dump while Soft can be dissipated by either dumping or passive mitigation.

It's a system I've thought for a while that would work very well in the Endless Sky balance system. I think it is a solution which is worth investigating as it is a non-immersion breaking, fun mechanic in another series which tackles the exact problem which currently exists with heat while simultaneously adding more knobs and variables for increased interaction.

**Comment by Amazinite** [comment]
To my understanding, flux is a more integral part of combat to Star Sector, whereas for us heat is something you can outright ignore if you have enough cooling and aren't fighting anyone with heat damage. And at the same time, the problems with heat damage are more to do with the player's ability to abuse it at higher levels and less to do with the AI using it against the player (although eventually someone might get sick of it and fight back). So I don't really see us emulating how Star Sector does flux by expanding how heat affects combat in Endless Sky.

**Comment by Ferociousfeind** [comment]
The flamethrower simply should be nerfed, period, to start with. It does 12,000 heat DPS, at the cost of 15 fuel per second, or 1 hyperjump in about 6 seconds. The suggested changes have been both to increase fuel consumption, and to decrease heat damage, but heat mechanics should be re-evaluated, yes.

I occasionally work on a branch that changes how heat works, making total shutdown a lot more difficult to achieve (I believe 200%?), but also in its stead more subtle and less absolute detrimental effects. Mainly, low-crew-like momentary loss of control, and a softening of hull leading to hull damage being gradually more effective against the structural-damage-accumulating ship.

**Comment by Quantumshark** [comment]
In-game, the flamethrower is stated to be mostly useful against small ships running close to their heat thresholds - the free worlds' anti-pirate development to mirror the plasma turret and dreadnought which are more designed to fight the Navy with their capital ships.
I don't like the idea of scaling heat damage to ship mass - it _should_ take more firepower to disable a capital ship than a smaller one, and that should be true of heat damage as it is true of any other damage type.
I think reducing the flamethrower's heat damage by 50% or even 75% wouldn't be unreasonable. Currently, it takes ~120 tons of outfit space (with only human equipment) to provide enough cooling to match the heat damage from a single 9 ton flamethrower.
I don't think heat damage is inherently more powerful than ion damage, its just that heat damage weapons currently do a lot more heat damage than ion damage weapons do ion damage - compare the plasma cannon to the ion cannon, for example, and the flamethrower even more so. An ion equivalent of the flamethrower would be just as overpowered as the flamethrower is currently.

**Comment by Amazinite** [comment]
Even outside of the flamethrower, heat damage is still a powerful damage type. Whereas ion damage slowly saps the energy of a ship and allows it to use less and less systems as ion damage increases up until the point of the ion damage being greater than the energy production, at which point nothing works, heat damage does nothing up until the point of reaching the heat capacity of a ship, and then it becomes completely disabled. There is no ramp up period for heat damage, just a sharp drop from functional to completely disabled and probably dead; so long as the attacker isn’t distracted, they can just keep pumping heat damage into a target and keep them disabled. And while a ship with more ion damage than energy generation is practically disabled, it’s systems still actually run, but a ship that is overheated is considered properly disabled and no systems, including cooling, will run, so an overheated ship is just relying on its passive heat dissipation in order to continue functioning.

**Comment by Maddremor** [comment]
Just spitballing, but couldn't the effects of overheating be scaled according to the gross mass of the ship? This would be due to a larger ship having it's mass and cargo act as a secondary fallback heat-sink, and that more actual thermal energy needed to be dumped into the system for it to reach truly catastrophic levels.

**Comment by Amazinite** [comment]
Well larger ships are already more heat resistant in requiring more heat damage to overheat, so I'm not to keen on also making them less impacted by overheating when it happens. I'd rather overheating simply be changed in some way across the board regardless of ship size.

**Comment by EugeneSanderson** [comment]
I also think the flame thrower should be nerfed. Having a carrier with a few remnant drones equipped with a flame thrower and fuel processors is seriously OP. It take mere seconds to push a capital ship to over 1000% heat. So far, this method can take out any ship, even the Arfecta with minimal effort

**Comment by Ferociousfeind** [comment]
okay so, for the flamethrower in specific, how are our thoughts on this adjustment? I believe that MZ's original intent for the flamethrower being so powerful (200 heat damage per frame!!!) was to burst for less than a second on small ships to disable them. We could enforce that intention by giving the flamethrower burst mechanics, as such:
```
outfit "Flamethrower"
	category "Secondary Weapons"
	cost 190000
	thumbnail "outfit/flamethrower"
	"mass" 9
	"outfit space" -9
	"weapon capacity" -9
	"gun ports" -1
	weapon
		sprite "effect/explosion/small"
			"frame rate" 60
			"no repeat"
		sound "flamethrower"
		icon "icon/flamethrower"
		"submunition" "Flamethrower Projectile"
		"inaccuracy" 12
		"velocity" 9
		"lifetime" 5
		"reload" 6
		"burst reload" 1
		"burst count" 20
		"firing energy" 1
		"firing fuel" .5
		"firing heat" 10
```
This is, in specific, addition of `reload 6` and `burst count 20`, which reduces the heat DPS from 12,000 to a much more reasonable 2000, while still giving bursts of 12,000 between two-second downtimes. `firing energy 1`, `firing fuel 0.5`, and `firing heat 10` , are 10x, 2x and 5x the original values. I feel like a sizable amount of energy consumption alongside more fuel consumption than there already is would make more sense. This is an experimental weapon, it is not a weapon of war refined over milennia of use and fine-tuning.

This doesn't solve the heat issue at large, but does put the flamethrower more in line with the other heat-producing outfits, so it isn't a sore thumb among sore thumbs

**Comment by Naminson** [comment]
Bit of a shot in the dark, but what if heat weapons heated the user up as much as the target? The flamethrower would probably be a much less obvious choice if it did 12,000 heat damage to **you** as well.

**Comment by Ferociousfeind** [comment]
doing exactly the same amount of heat damage to yourself as to your opponent would render them effectively useless, especially since not every shot is guaranteed to hit. But you're right, the flamethrower should incur a lot more firing heat.

**Comment by Amazinite** [comment]
#6799 buffed ion damage by giving it a new mechanic where it can cause weapons to "jam," thus lowering the DPS of the target, but even with bringing ion damage up, heat damage is still rather powerful. But the core of heat damage is the overheat mechanic, so really what we should be doing is looking at making the overheat mechanic less punishing.

Azure on Discord suggested changing overheating so that an overheated ship, instead of becoming completely disabled, only has its power generation and engines disabled. This way an overheated ship can still fight using its turrets, for so long as its energy stores hold up.

I like this approach to changing overheating, but think we should ditch the disabled engines idea; impacting engines is the realm of slowing damage, so I don't want heat damage to also impact engines. But the idea of overheating disabling *only* power generation sounds like it could work.

So the change would be that instead of overheated ships being unable to act (currently an overheated ship is almost entirely synonymous with a disabled ship), overheated ships shut down their reactors. This means that all `"energy generation"` and `"heat generation"` would cease. The disabling of heat generation would actually help the ship return from being overheated quicker, but an overheated ship would only be able to function for so long as it still has battery power remaining. Only once the ship is out of battery power would it reach a state similar to what we have now. So while overheating would still be powerful, there'd at least be a way to combat it outside of stacking so much cooling that your idle heat is massively negative and you never overheat in the first place. And an overheated ship could still potentially escape or fight back using what battery power remains.

Along with disabling energy generation, overheated ships could perhaps also have disabled shield and hull repair. This would save the ship some energy, but make overheated ships softer targets.

Solar power, instead of being completely disabled by being overheated, could perhaps simply be diminished. In real life, solar panels lose efficiency at higher temperatures, so solar power in game losing effectiveness while the ship is overheated could be analogous to that. So while solar power wouldn't be a pure hard counter to overheating, it would be a way to mitigate the effects of being overheated by being able to hold onto at least some power generation.

**Comment by ThrawnCA** [comment]
> Only once the ship is out of battery power would it reach a state similar to what we have now. So while overheating would still be powerful, there'd at least be a way to combat it outside of stacking so much cooling that your idle heat is massively negative and you never overheat in the first place. And an overheated ship could still potentially escape or fight back using what battery power remains.

Wouldn't that mean you could become almost immune to heat damage by stacking up a few Remnant crystal capacitors? For less than the outfit space of a Liquid Helium Cooler, you could have enough energy storage to simultaneously run an A860 Atomic Thruster, A865 Atomic Steering, four Heavy Lasers, and four Heavy Laser Turrets, for 20 seconds, regardless of overheating.

**Comment by Amazinite** [comment]
Crystal capacitors draw energy, so even though they have a high energy capacity density, you'd be losing that energy even faster when overheated.

**Comment by ThrawnCA** [comment]
> Crystal capacitors draw energy, so even though they have a high energy capacity density, you'd be losing that energy even faster when overheated.

It's a trivial drain compared to the capacity, though. Maybe enough to lower the 20-second full-combat duration to 19.9 seconds.

**Comment by the2nake** [comment]
Could a possible solution to be to make overheating reduce the efficiency of your outfits, instead of relying on remaining power or completely disabling certain sections?

**Comment by Ferociousfeind** [comment]
If I may add, I would vastly prefer outfits to individually shut down after their local heat tolerances have been exceeded, so we can make outfit-by-outfit decisions on what attributes stop when you overheat, rather than a game-wide decision. All heat generation and all energy generation stops when your heat goes above 100%, even when in-universe that is literally impossible (think RTGs which produce power via passive nuclear decay instead of a triggered nuclear reaction)

Certain outfits are hardy and can take a beating. Certain outfits opt out of the whole "heat makes me break" nonsense, and Certain outfits are fragile and need to be kept cool (or you'll void the warrantee).

With certain implementations of outfit-specific overheat behavior, you could have problem players like archons and affects simply opt out of heat entirely, make them impervious to heat because they're too high-tech to care about your pitiful human gas-blower. You could have engined which shut down when you're too hot, and others which don't do that. You could have powerful reactors which shut down, and also other reactors which are less space-efficient but don't buckle to high heat environments. You could have shield generators which shut down at 90% heat, or cloaks which stop if you're above 50% heat, or guns which jam if you're above 300% heat.

I dislike the galaxy-wide mandate to only produce power generators which shut off if they're above 100% heat. Reactors which can't do that _can_ exist, and if descriptions are anything to go off of, _humans have them already._

**Comment by EugeneSanderson** [comment]
Or add a sterling type engine that increases in efficiency when the ship heats up,. This would be considered a hard counter at the cost of outfit space. 
But... Here is the catch. The engine is not effective around hot stars when the ship is hot, because the heat cannot escape, though it is extremely effective far away from hot stars as heat is converted into energy at greater efficiencies -  Carnot heat cycle efficiency

**Comment by EugeneSanderson** [comment]
Another hard counter idea is that one can use fuel to cool down the ship. Since the fuel is hydrogen, one can add a fuel decompression unit. Hydrogen will be in the form of a liquid and once expanded it will cool down the ship at a massive rate, but will take time and fuel to recharge (compress to liquid form again)

**Comment by Zitchas** [comment]
I like those ideas, to particularly having variable heat impacts as Ferocious fiend describes. We already have more than a few outfits that include heat descriptors, so whole this heat disabling may be the norm, our lore already supports exceptions or at least variations on this.
